overlord/devicestate/devicemgr.go

```diff
@@ -435,6 +435,26 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+var startTime time.Time
+
+func init() {
+	startTime = time.Now()
+}
+
+func (m *DeviceManager) setTimeOnce(name string, t time.Time) error {
```

Comment:
Could this be a helper on state, eg. state.SetOnce("key", value) ?

Reply:
it certainly cloud be, @pedronis what do you think ( and also note that like the other comment this really applies to #9035 not this PR)

Reply:
it seems an ok idea, but maybe follow up material,  it's also the first use case for this, so not sure it's worth adding to state yet.